justfile

```diff
@@ -0,0 +1,58 @@
+alias e := export
+alias ea := export-all
+alias r := refresh
+alias ra := refresh-all
+alias u := update
+alias ua := update-all
+
+set windows-shell := ["powershell.exe", "-NoLogo", "-Command"]
+
+default:
+    @just --list
+
+[private]
+[windows]
+_batchcmd cmd loader:
+    Get-ChildItem -Path versions\{{ loader }} -Directory | % { \
+      Set-Location $_.FullName; \
+      Write-Host "running {{ cmd }} in $_.FullName"; \
+      Invoke-Expression "{{ cmd }}"; \
+      Pop-Location; \
+    }
+
+[linux]
+[macos]
+[private]
+_batchcmd cmd loader:
+    for d in versions/{{ loader }}/*/; do \
+      cd "$d"; \
+      echo "running {{ cmd }} in $d..."; \
+      {{ cmd }}; \
+      cd {{ invocation_directory() }}; \
+    done
+
+# all versions of <loader> will be exported as a modrinth modpack
+[linux]
+[macos]
+export loader: && (_batchcmd "packwiz modrinth export" loader)
+    -mkdir -p .build/{{ loader }}
+
+# all versions of <loader> will be exported as a modrinth modpack
+[windows]
+export loader: && (_batchcmd "packwiz modrinth export" loader)
+    -New-Item -Type Directory -Path .build\{{ loader }}
+
+# all versions of <loader> will have pack.toml refreshed
+refresh loader: && (_batchcmd "packwiz refresh" loader)
+
+# all versions of <loader> will be updated
+update loader: && (_batchcmd "packwiz update --all" loader)
+
+# refresh all modpacks
+refresh-all: (refresh "fabric") (refresh "quilt")
+
+# update all modpacks
+update-all: (update "fabric") (update "quilt")
```

Comment:
Why not add an exception for the `all` keyword to _batchcmd instead of this?

It can just be `refresh all` or `update all` etc then.

(not sure how it would work for windows since I"m not familiar with it

Reply:
Also you can glob the modloaders (for forge for example, or whenever neoforge comes out)

Reply:
done! thanks :)